### DISCUSSION TITLE
About the animation features in PyDPainter

### DISCUSSION THREAD

**Discussion by MetalMaxMX** [mriale/PyDPainter#81] [General]
I have noticed that, within the roadmap here, there's work slated to have animation features 
https://github.com/mriale/PyDPainter/wiki/RoadMap

I am kinda curious about that. The roadmap stated that they would be similar in behavior to DPaint's II capabilities, but there's also this offshoot that was meant exclusively for animation as well (https://en.wikipedia.org/wiki/Deluxe_Paint_Animation) 

Was there any noticeable difference between them? And what would it take for PyDPainter to have those animation features in general? (either from DPII or DPA)

I am asking here because I've seen a few documentaries about Amiga games and they tended to experiment a lot with palette cycling, morphing and more so it'd be cool to know how they get applied here. 

Cheers!
MetalMaxMX

**Comment by mriale** [comment]
The version numbers on the roadmap page refer roughly to the Amiga Deluxe Paint version numbers. Deluxe Paint for DOS was a different thing. Some things have been pulled in from it like the chunky file format, but the main goal is to recreate the Amiga version.

The project is currently at version 1.2.1 which means PyDpainter has more functionality than the original DPaint but it's not quite up to the Amiga version of DPaint II yet. These things need to be added to get there:
* 1.3 Perspective mode
* 1.4 Amiga font support

The Amiga version of DPaint III added animation, so the 2.x versions will start to add animation features starting out with basic setting number of frames and moving among them. I'll also have to implement IFF ANIM and animated GIF import/export. Animbrush and sprite sheet functionality would come next. Then the move requestor, which is one of the major features that Deluxe Paint III added for supporting animation. Of course that depends on Perspective mode, which should have been done in version 1.3.

I expect to start working on the animation features before the end of 2023, but of course, at this point I'm only one person doing this in my spare time.

By the way, color cycling is already implemented in PyDPainter if you want to play with that functionality. In my quick demo video I show color cycling:
https://www.youtube.com/watch?v=GGfsSvbrb4o

Hope that helps!
Mark Riale

**Reply by MetalMaxMX** [reply]
Hey thanks for answering! And specially so for the color cycling functionality in DPainter. I just would like to know though, since I get confused here and there. Layers = Animation, right? I have seen that terminology being used when referring to the animation of one or multiple sprites at once, and I wanted to know if it's a similar term used on the Amiga (I know GrafX2 for instance uses the term layers, while other applications may use additional windows or so to refer to them as frames)

**Reply by mriale** [reply]
Deluxe Paint never really had the concept of layers. The animation features allowed you to paint on individual frames or have motions that zoom and rotate brushes. This clip from back in the day shows these features:
https://www.youtube.com/watch?v=WjIQO1MjV2w

**Reply by MetalMaxMX** [reply]
Ah, thanks once more! The video was rather interesting. Though makes me wonder how these animations were stored and then used in games or other demos. You mentioned IFF ANIM and GIF, were those the formats/methods used back then?

**Reply by mriale** [reply]
Deluxe Paint used IFF ANIM to store the compressed differences between the frames to make the files smaller. It didn't use animated GIFs.

**Reply by MetalMaxMX** [reply]
Ahh, I see! Makes sense back then to have a specific type of format given the limitations in memory that the Amiga had. Well, that's all from me. So huh, I hope more things can be announced down the line and if more updates on the animation features on PyDPainter appear, that would be welcome! 

Once again, thank you very much for answering my queries. I am particularly excited by both the animation and sprite sheet functionality (didn't know the Amiga had one!) coming at some point so I'll be on the lookout for it!

**Comment by mriale** [comment]
@MetalMaxMX FYI, I don't know if you saw, but I released 2.0 a few days ago and then a quick bug fix to 2.0.1 today.

2.0 Trailer video:
https://youtu.be/l385Z7_CRB0

Download link:
https://pydpainter.org/download.php

Still haven't implemented anim brushes and sprite sheets yet, though.

**Reply by MetalMaxMX** [reply]
Hello, yes I've heard of it and I already got it working and funkin' on my end :) 

If I had to make one bit suggestion though would be to mention a dependency (I believe it was python tk or something) because the crash error GUI report uses that and that isn't indicated anywhere on the site. I ended up getting a crash somehow but I don't recall how to reproduce it sadly, not sure what went wrong in that one really. 

Otherwise, congratulations on the 2.0 release and looking forward to anim brushes and sprite sheets! (And also the layer functionality which I've seen being mentioned on Github)

**Reply by mriale** [reply]
Added issue #164 to document tkinter dependency.
